### PR TITLE
added customer entity, migrations and GetAllCustomers query, endpoint…

### PR DESCRIPTION
**Pull request by fedordavid**
created new customer entity, executed migrations, created unit,integration tests, created getAllCustomers query and controller api endpoint

### CODE REVIEW

File: freddytoolapi/Freddy.IntegrationTests/CustomWebApplicationFactory.cs [hunk 1]
```diff
@@ -43,6 +43,7 @@ protected override IHost CreateHost(IHostBuilder builder)
             {
                 var context = scope.ServiceProvider.GetService<DatabaseContext>();
                 new Products().Initialize(context);
+                new Customers().Initialize(context);
                 //context.Database.EnsureDeleted();
                 context.Database.Migrate();
```
Comment (TomasDurica) [inline freddytoolapi/Freddy.IntegrationTests/CustomWebApplicationFactory.cs:46]: oops, ta migracia by mala byt nad inicializaciou :D

File: freddytoolapi/Freddy.API/Controllers/OrderController.cs [hunk 2]
```diff
@@ -54,5 +56,11 @@ public async Task<ActionResult> UpdateProduct(Guid productId, ProductInfo produc
             await _commandBus.Handle(new UpdateProductCommand(productId, productInfo));
             return Ok();
         }
+
+        [HttpGet("api/freddy/customers")]
+        public async Task<ActionResult<CustomerView[]>> GetAllCustomers()
+        {
+            return await _queryBus.Execute(new GetAllCustomersQuery());
+        }
     }
```
Comment (TomasDurica) [inline freddytoolapi/Freddy.API/Controllers/OrderController.cs]: Mozno rozdelit do viacerych controllerov? (ProductController a CustomerController) - ale to mozme aj v refaktoringu

File: freddytoolapi/Freddy.IntegrationTests/Controllers/OrderControllerTests.cs [hunk 3]
```diff
@@ -157,6 +158,27 @@ await using (var ctx = CreateDatabaseContext())
...
+
+        [Fact]
+        public async Task GetAllCustomers_ShouldReturnDataFromDatabase()
+        {
+            const string url = "api/freddy/customers";
+
+            var result = await _client.GetObjectAsync<CustomerView[]>(url);
+
+            Assert.Equal(2, result.Length);
+            // TODO: Compare with TestData
```
Comment (TomasDurica) [inline freddytoolapi/Freddy.IntegrationTests/Controllers/OrderControllerTests.cs]: Mozes to TODO aj to nad tym vyriesit ;)

**Review by TomasDurica — APPROVED** [review]

File: freddytoolapi/Freddy.IntegrationTests/CustomWebApplicationFactory.cs (hunk 1 above)
Comment (fedordavid) [reply freddytoolapi/Freddy.IntegrationTests/CustomWebApplicationFactory.cs:46]: Otocil som , u mna najprv neprechadzali testy takze som odkomentoval EnsureDeleted a potom to preslo ok nasledne zakomentoval

File: freddytoolapi/Freddy.API/Controllers/OrderController.cs (hunk 2 above)
Comment (fedordavid) [reply freddytoolapi/Freddy.API/Controllers/OrderController.cs]: Vyriesene akurat trebalo na testy este [Collection("Integration")] , takze diki za tip

File: freddytoolapi/Freddy.IntegrationTests/Controllers/OrderControllerTests.cs (hunk 3 above)
Comment (fedordavid) [reply freddytoolapi/Freddy.IntegrationTests/Controllers/OrderControllerTests.cs]: Dorobim get products z db a v cykle foreach assertujem, neviem ci to je najlepsi aproach ale mozes potom navrhnut ked commitnem, zatial sa zda ze funguje ok

**Review by TomasDurica — APPROVED** [review]

**Pull request merged**